pkg/agent/endpoints/endpoints.go

```diff
@@ -107,3 +114,19 @@ func (e *endpoints) createUDSListener() (net.Listener, error) {
 	os.Chmod(e.c.BindAddr.String(), os.ModePerm)
 	return l, nil
 }
+
+func UnknownServiceHandler(log logrus.FieldLogger) grpc.StreamHandler {
+	const sdsMethodPrefix = "/envoy.service.discovery.v2.SecretDiscoveryService/"
+	logged := false
+	return func(srv interface{}, stream grpc.ServerStream) error {
+		method, ok := grpc.MethodFromServerStream(stream)
+		if ok && strings.HasPrefix(method, sdsMethodPrefix) {
+			if !logged {
+				log.Warn("Incoming RPC for Envoy SDS but it is not enabled")
+				logged = true
+			}
+			return status.Error(codes.Unimplemented, "Envoy SDS support has not been enabled on the agent")
```

Comment:
nit: consider a doc link, or naming the config flag. Might be able to save a step.

Reply:
grand idea!